src/storage/src/hummock/local_version/local_version_manager.rs

```diff
@@ -47,21 +47,23 @@ use crate::hummock::shared_buffer::OrderIndex;
 use crate::hummock::sstable_store::SstableStoreRef;
 use crate::hummock::utils::validate_table_key_range;
 use crate::hummock::{
-    HummockEpoch, HummockError, HummockResult, HummockVersionId, MemoryLimiter,
-    SstableIdManagerRef, TrackerId, INVALID_VERSION_ID,
+    HummockEpoch, HummockError, HummockResult, MemoryLimiter, SstableIdManagerRef, TrackerId,
+    INVALID_VERSION_ID,
 };
 #[cfg(any(test, feature = "test"))]
 use crate::monitor::StateStoreMetrics;
 use crate::storage_value::StorageValue;
 use crate::store::SyncResult;
 
 struct WorkerContext {
-    version_update_notifier_tx: tokio::sync::watch::Sender<HummockVersionId>,
+    version_update_notifier_tx: tokio::sync::watch::Sender<HummockEpoch>,
 }
 
 impl WorkerContext {
-    pub fn notify_version_id(&self, version_id: HummockVersionId) {
-        self.version_update_notifier_tx.send(version_id).ok();
+    pub fn notify_max_committed_epoch(&self, max_committed_epoch: HummockEpoch) {
+        self.version_update_notifier_tx
+            .send(max_committed_epoch)
+            .ok();
```

Comment:
> This ignores the send error slightly. According to the watch channel doc (https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html#method.send), `send` can fail when there are no receivers, which is the case when there is no one calling `try_wait_epoch`. This can cause try_wait_epoch to take longer than expected.
> 
> Example:
> 
> ```
> 1. Epoch is updated from e2 -> e3
> 2. try_wait_epoch e3 is called.
> 3. Epoch is updated from e3 -> e4.
> ```
> 
> In the above example, `notify_max_committed_epoch` will fail in 1) and 2) can only complete when 3) is finished.

In his latest code, `wait_epoch` will read local_version first, so this situation will not happen.